Posh-ACME.Deploy/Private/Confirm-CertInstall.ps1

```diff
@@ -17,17 +19,11 @@ function Confirm-CertInstall {
             throw "CertThumbprint and PfxFile were not provided. You must specify one or both of them."
         }
 
-        # grab the cert thumbprint from the PFX file if it wasn't specified
-        if (-not $CertThumbprint) {
-            Write-Verbose "Attempting to read Pfx thumbprint"
-            $CertThumbprint = Get-PfxThumbprint -PfxFile $PfxFile -PfxPass $PfxPass
-        }
-
         # install the cert if necessary
         if (-not (Test-CertInstalled $CertThumbprint)) {
-            if ($PfxFile) {
+            if ($PfxFile) {# grab the cert thumbprint from the PFX file if it wasn't valid/installed
                 $PfxFile = $ExecutionContext.SessionState.Path.GetUnresolvedProviderPathFromPSPath($PfxFile)
-                Import-PfxCertInternal $PfxFile -PfxPass $PfxPass
+                $CertThumbprint = Import-PfxCertInternal $PfxFile -PfxPass $PfxPass

```

Comment:
Because `Test-CertInstalled` now allows for empty thumbprints, it's possible the Import function would get called for a Pfx file that already has a copy of the cert installed, right? I just tested the `Import-PfxCertInternal` with a duplicate and it doesn't seem to throw any errors. But I may end up throwing an already-installed check in the function as well once it has the thumbprint.

Reply:
Well maybe.  There are two places the cert needs to exist before the script which is calling the internal Confirm-CertInstall.ps1 script.  

1) The certificate store (which is where this routine is looking
AND
2) The meta data the calling script is trying to configure (IIS, Exchange, Remote Access, etc.)

Just because the cert was previously contained the certificate store already DOES NOT mean that the meta-data is properly referencing it.  It is up to the calling script to manage its meta data...

Reply:
Right. I guess what I mean is...the only thing calling `Import-PfxCertInternal` right now is `Confirm-CertInstall` which tries to check if the cert is already installed to the cert store via the thumbprint. But if it gets a null or empty `CertThumbprint` parameter, it determines the cert isn't installed (even if it actually is) and calls `Import-PfxCertInternal`. So I'm thinking it might make sense to just move that `Test-CertInstalled` check into `Import-PfxCertInternal` after it has read the PFX and has the thumbprint to do a real check.

Reply:
Currently (in what I've proposed) Test-CertInstalled is only called from Confirm-CertInstall and only if the specified cert isn't installed is the Import-PfxCertInternal script called,  I'm not sure I follow your thinking.

The input thumbprint may have been empty or completely incorrect but a valid PFX and password has been supplied.  If the provided thumbprint is good, we're done otherwise,  The valid PFX should be used and whatever its thumbprint is should be what is ultimately referenced by the caller.
Given that a useful PFX was supplied which may or may not have actually previously been in the cert store why would it matter to the caller?  It just wants to be assured that the cert is in the store and that it is using the correct thumbprint.

Reply:
You're correct it doesn't matter to the caller and nothing is broken as-is. I was just thinking in terms of efficiency. When only a PFX is supplied by the user, `Confirm-CertInstall` calls `Test-CertInstalled` with an empty thumbprint which always returns false. `Import-PfxCertInternal` then always imports the PFX even if a copy may already exist because it wasn't able to properly check if it existed first. I'm suggesting moving (or maybe just copying) the `Test-CertInstalled` check into `Import-PfxCertInternal` after it reads the PFX file (and can get the thumbprint) but before it actually writes the data to the cert store. Or perhaps even collapsing all the functionality into `Confirm-CertInstall`.

Reply:
This might make more sense if I just tweak things after merging this PR. I'm honestly ok with it as-is and I sincerely appreciate the time you've put into it.

Reply:
That's a good idea.  

I thought of merging in the Test logic when I added the one line to return false from Test-CertInstalled with a null thumbprint, but removing this seemed a little aggressive and it might prove useful in the future.

There are places where worrying about efficiency is absolutely important, but in general, these deployment activitieswon't be invoked more often than once per month so the small fraction of a second gain can't matter much...